images/base/entrypoint

```diff
@@ -42,6 +42,31 @@ fix_mount() {
   mount --make-rshared /
 }
 
+fix_cgroup() {
+  echo 'INFO: fix cgroup mounts for all subsystems'
+  # For each cgroup subsystem, Docker does a bind mount from the current
+  # cgroup to the root of the cgroup subsystem. For instance:
+  #   /sys/fs/cgroup/memory/docker/<cid> -> /sys/fs/cgroup/memory
+  #
+  # This will confuse Kubelet and cadvisor and will dump the following error
+  # messages in kubelet log:
+  #   `summary_sys_containers.go:47] Failed to get system container stats for ".../kubelet.service"`
+  #
+  # This is because `/proc/<pid>/cgroup` is not affected by the bind mount.
+  # The following is a workaround to recreate the original cgroup
+  # environment by doing another bind mount for each subsystem.
+  MOUNT_TABLE=$(cat /proc/self/mountinfo)
+  DOCKER_CGROUP_MOUNTS=$(echo "${MOUNT_TABLE}" | grep /sys/fs/cgroup | grep docker)
```

Comment:
we should be resilient to not running in docker, we're nearing support for eg podman.

Reply:
Hum, ok, i don't know that. I haven't tested podman... what do you think about adding some check (e.g., checking .dockerenv) and only run this for docker?

Reply:
do we have to do this from the entrypoint? 
can we do this from https://github.com/kubernetes-sigs/kind/blob/master/pkg/internal/cluster/providers/docker/provision.go#L147?

the docker cgroup folder is based in the container id, right?

Reply:
Agree with @aojea . Moving these logics to provides/docker can be clearer.

Reply:
it should happen before we start systemd, entrypoint is appropriate.
this is roughly the right approach, but we can make the bash slightly more clever.

I'll look into this more today, I had to handle some other pressing issues yesterday.

Reply:
Do you know if using the `cgroup-driver=systemd` as _kubeadm_ extra options could fix the issue?

[Doc](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-control-plane-node) here

Reply:
no, the driver should match the container runtime. it does.

https://kubernetes.io/docs/setup/production-environment/container-runtimes/#systemd

Reply:
kubelet and the CRI need to agree on this, they do. IIRC we tried using the systemd driver but it did not help, we can try this again, but it needs to be both kubelet and containerd.

Reply:
I think we can deal with podman later, however I've made some minor tweaks in additional commit to the bash